packages/iov-core/src/writer.ts

```diff
@@ -34,12 +52,12 @@ export class IovWriter {
   }
 
   public reader(chainId: ChainId): BcpConnection {
-    return this.getChain(chainId).client;
+    return this.getChain(chainId).connection;
   }
 
   public async addChain(connector: ChainConnector): Promise<void> {
     const connection = await connectChain(connector);
-    const chainId = connection.client.chainId();
+    const chainId = connection.connection.chainId();
```

Comment:
Hmmm... a bit of stutter. I like the renaming, but maybe this could be clearer.
Just a thought, no need to change this for the pr